src/components/KYCWall/BaseKYCWall.tsx

```diff
@@ -0,0 +1,280 @@
+            }
+
+            // Use event target as fallback if anchorRef is null for safety
+            const targetElement = anchorRef.current ?? (event?.nativeEvent.target as Element);
+
+            transferBalanceButtonRef.current = targetElement;
+            const isExpenseReport = ReportUtils.isExpenseReport(iouReport);
+            const paymentCardList = fundList ?? {};
+
+            // Check to see if user has a valid payment method on file and display the add payment popover if they don't
+            if (
+                (isExpenseReport && (reimbursementAccount?.achData?.state ?? '') !== CONST.BANK_ACCOUNT.STATE.OPEN) ||
+                (!isExpenseReport && !PaymentUtils.hasExpensifyPaymentMethod(paymentCardList, bankAccountList, shouldIncludeDebitCard))
+            ) {
+                Log.info('[KYC Wallet] User does not have valid payment method');
+                if (!shouldIncludeDebitCard) {
+                    selectPaymentMethod(CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT);
+                    return;
+                }
+
+                const clickedElementLocation = getClickedTargetLocation(targetElement);
+                const position = getAnchorPosition(clickedElementLocation);
+
+                setPositionAddPaymentMenu(position);
+                setShouldShowAddPaymentMenu(true);
+
+                return;
+            }
+
+            if (!isExpenseReport) {
+                // Ask the user to upgrade to a gold wallet as this means they have not yet gone through our Know Your Customer (KYC) checks
+                const hasActivatedWallet = userWallet?.tierName && [CONST.WALLET.TIER_NAME.GOLD, CONST.WALLET.TIER_NAME.PLATINUM].includes(userWallet.tierName);
+                if (!hasActivatedWallet) {
+                    Log.info('[KYC Wallet] User does not have active wallet');
+                    Navigation.navigate(enablePaymentsRoute);
+                    return;
+                }
+            }
+            Log.info('[KYC Wallet] User has valid payment method and passed KYC checks or did not need them');
+            onSuccessfulKYC(currentSource, iouPaymentType);
+        },
+        [
+            bankAccountList,
+            chatReportID,
+            enablePaymentsRoute,
+            fundList,
+            getAnchorPosition,
+            iouReport,
+            onSuccessfulKYC,
+            reimbursementAccount?.achData?.state,
+            selectPaymentMethod,
+            shouldIncludeDebitCard,
+            shouldShowAddPaymentMenu,
+            source,
+            userWallet?.tierName,
+            walletTerms?.source,
+        ],
+    );
+
+    useEffect(() => {
+        let dimensionsSubscription: EmitterSubscription | null = null;
+
+        PaymentMethods.kycWallRef.current = {
+            continueAction: (event?: SyntheticEvent<NativeTouchEvent>, iouPaymentType?: TransferMethod) => {
+                continueAction(event, iouPaymentType);
+            },
+        };
```

Comment:
1) Can it be simplified?
`PaymentMethods.kycWallRef.current = {continueAction};`
2) Please make sure it works the same way as before